src/models/outbound/pispOTPValidate.model.ts

```diff
@@ -0,0 +1,236 @@
+
+  get thirdpartyRequests (): ThirdpartyRequests {
+    return this.config.thirdpartyRequests
+  }
+
+  static notificationChannel (consentRequestsRequestId: string): string {
+    if (!consentRequestsRequestId) {
+      throw new Error('PISPOTPValidateModel.notificationChannel: \'consentRequestsRequestId\' parameter is required')
+    }
+    // channel name
+    return `PISPOTPValidate-${consentRequestsRequestId}`
+  }
+
+  async onValidateOTP (): Promise<void> {
+    const { consentRequestsRequestId, authToken, toDFSPParticipantId } = this.data
+
+    const channel = PISPOTPValidateModel.notificationChannel(
+      consentRequestsRequestId
+    )
+
+    this.logger.push({ channel }).info('onValidateOTP - subscribe to channel')
+
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(async (resolve, reject) => {
+      let subId = 0
+      try {
+        // this handler will be executed when POST /consents @ Inbound server
+        subId = this.pubSub.subscribe(channel, async (channel: string, message: Message, sid: number) => {
+          // first unsubscribe
+          this.pubSub.unsubscribe(channel, sid)
+
+          type PutResponseOrError = tpAPI.Schemas.ConsentsPostRequest & fspiopAPI.Schemas.ErrorInformationObject
+          const putResponse = message as unknown as PutResponseOrError
+          if (putResponse.errorInformation) {
+            this.data.errorInformation = putResponse.errorInformation
+          } else {
+            this.data.consent = { ...message as unknown as tpAPI.Schemas.ConsentsPostRequest }
+          }
+          resolve()
+        })
+
+        const res = await this.thirdpartyRequests.patchConsentRequests(
+          consentRequestsRequestId,
+          { authToken: authToken }  as tpAPI.Schemas.ConsentRequestsIDPatchRequest,
+          toDFSPParticipantId
+        )
+        this.logger.push({ res }).info('ThirdpartyRequests.patchConsentRequests request sent to peer')
+      } catch (error) {
+        this.logger.push(error).error('ThirdpartyRequests.patchConsentRequests request error')
+        this.pubSub.unsubscribe(channel, subId)
+        reject(error)
+      }
+    })
+  }
+
+  getResponse ():
+  OutboundOTPValidateConsentResponse |
+  OutboundOTPValidateErrorResponse |
+  void {
+    switch (this.data.currentState) {
+      case 'OTPIsValid':
+        return {
+          consent: this.data.consent,
+          currentState: this.data.currentState
+        } as OutboundOTPValidateConsentResponse
+      case 'errored':
+        return {
+          errorInformation: this.data.errorInformation,
+          currentState: this.data.currentState
+        } as OutboundOTPValidateErrorResponse
+      default:
+    }
+  }
+
+  // utility function to check if an error after a transistion which
+  // pub/subs for a response that can return a mojaloop error
+  async checkModelDataForErrorInformation(): Promise<void> {
+    if (this.data.errorInformation) {
+      await this.fsm.error()
```

Comment:
I would give it a try:
```suggestion
      await this.fsm.error(this.data.errorInformation)
```

Reply:
done.